[DEVX-1701] Add docs for saucectl timeout cli output (#1277)
saucelabs/sauce-docs: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md

Instructs how long `saucectl` should wait for the suite to complete, potentially overriding the default project timeout setting.

> Once the suite hits the timeout limit, its status is set to `?` within the CLI. This does not reflect the actual status of the job you may see in the Sauce Labs Web UI or API.

Reviewer: ```suggestion
When the suite reaches the timeout limit, its status is set to '?' in the CLI. This does not reflect the actual status of the job in the Sauce Labs web UI or API.
```
